port/zephyr/CMakeLists.txt

```diff
@@ -5,6 +5,23 @@ endif()
 zephyr_include_directories(include)
 zephyr_include_directories(../../src/include)
 
+function (include_cert path name)
+
+  if(NOT IS_ABSOLUTE ${path})
+    if(EXISTS ${APPLICATION_SOURCE_DIR}/${path})
+      set(path ${APPLICATION_SOURCE_DIR}/${path})
+    elseif(EXISTS ${CMAKE_CURRENT_SOURCE_DIR}/${path})
+      set(path ${CMAKE_CURRENT_SOURCE_DIR}/${path})
+    else()
+      message(FATAL_ERROR "${path} does not exist")
```

Comment:
That would require that this function take the name of the Kconfig option just to use it in a debugging message, and I think it's less clean to add an argument just for that. Also, right now this function doesn't assume that the path is ultimately set by a Kconfig option, and I prefer to keep that flexibility. Ultimately, I don't think it's that hard to work backwards from the path to Kconfig option, especially because users are likely to encounter this error just after setting the option.

Reply:
Below is the error you get when the options aren't set. Because the argument is completely blank CMake interprets it as whitespace and throws an error for too few arguments even before entering the function. That means that there's no way to have an error message inside the function for that case. It does however, show you the line that calls that function, which would then show you the Kconfig option that wasn't set. So I still think it's pretty easy to get to the root cause here without additional error messages.

```
CMake Error at 
/Users/samfriedman/golioth/test_cp_sdk_zephyr/modules/lib/golioth-firmware-sdk/examples/zephyr/common/CMakeLists.txt:8 
(include_cert):
  include_cert Function invoked with incorrect arguments for function named:
  include_cert


CMake Error at 
/Users/samfriedman/golioth/test_cp_sdk_zephyr/modules/lib/golioth-firmware-sdk/examples/zephyr/common/CMakeLists.txt:9 
(include_cert):
  include_cert Function invoked with incorrect arguments for function named:
  include_cert
```

Reply:
Updated to take in a variable name instead of a path.